### PR TITLE
Update Hugging Face Doc

### PR DESCRIPTION
**Pull request by gary149**
## Update Hugging Face documentation

Documentation was outdated for our current offering.

## Type

📖 Documentation

## Changes

1. Updated introduction to focus on Serverless Inference API and Dedicated Inference Endpoints.
2. Reorganized usage examples into these two main categories.
3. Added Serverless Inference API examples using meta-llama/Meta-Llama-3.1-8B-Instruct model.
4. Updated proxy configuration for Serverless API to use llama-3.1-8B-instruct model.
5. Added instructions and examples for Dedicated Inference Endpoints, including proxy configuration.
6. Updated streaming example to use Meta-Llama-3.1-8B-Instruct model.
7. Removed section on "Text Generation (NOT TGI)" models. (outdated)
8. Improved overall formatting and clarity.

### CODE REVIEW

File: docs/my-website/docs/providers/huggingface.md [hunk 1]
```diff
@@ -4,42 +4,42 @@ import TabItem from '@theme/TabItem';
 
 # Huggingface
 
-LiteLLM supports the following types of Huggingface models:
-* Text-generation-interface: [Here's all the models that use this format](https://huggingface.co/models?other=text-generation-inference).
```
Comment (ghost) [inline docs/my-website/docs/providers/huggingface.md:8]: It's okay to call out the dedicated endpoints, but it's more useful for me to know what the api formats supported are. 

TGI is an API format, so it's a useful callout.

Comment (gary149) [reply docs/my-website/docs/providers/huggingface.md:8]: Ok I updated, fyi all Text-generation models via HF Inference now use TGI. And conversational-task is depreciated.

**Comment by ishaan-jaff** [comment]
good to merge @gary149 @krrishdholakia ?

**Comment by gary149** [comment]
LGTM

**Pull request merged**